### PR TITLE
client/http: fix the scan region parameter

### LINKED ISSUE #7869: Bugs of using pd http client to scan regions
### What did you do?

use http pd client to scan regions

### What did you expect to see?
get correct regions

### What did you see instead?
1. client will panic by using `http.NewClient`, becase `client.serviceModeUpdateCb` is null by default
2. can't get the right regions in range because of different http format

### What version of PD are you using (`pd-server -V`)?
v6.5.3

### PR DESCRIPTION
**Pull request by Smityz**
### What problem does this PR solve?

Issue Number: Close #7869

### What is changed and how does it work?

```commit-message
add null pointer check and fix http request format
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by rleungx — APPROVED** [review]

File: client/http/api.go [hunk 1]
```diff
@@ -95,7 +95,7 @@ func RegionByKey(key []byte) string {
 // RegionsByKeyRange returns the path of PD HTTP API to scan regions with given start key, end key and limit parameters.
 func RegionsByKeyRange(keyRange *KeyRange, limit int) string {
 	startKeyStr, endKeyStr := keyRange.EscapeAsUTF8Str()
-	return fmt.Sprintf("%s?start_key=%s&end_key=%s&limit=%d",
+	return fmt.Sprintf("%s?key=%s&end_key=%s&limit=%d",
```
Comment (JmPotato) [inline client/http/api.go:98]: Would you like to add a test for this API in `tests/integrations/client/http_client_test.go` to ensure it works as expected?

Comment (JmPotato) [inline client/http/api.go:98]: Would you like to add a test for this API in `tests/integrations/client/http_client_test.go` to ensure it works as expected?

Comment (Smityz) [reply client/http/api.go:98]: sure

File: tests/integrations/client/http_client_test.go [hunk 2]
```diff
@@ -204,6 +204,12 @@ func (suite *httpClientTestSuite) checkMeta(mode mode, client pd.Client) {
 	version, err := client.GetClusterVersion(env.ctx)
 	re.NoError(err)
 	re.Equal("0.0.0", version)
+	rgs,_:=client.GetRegionsByKeyRange(env.ctx, pd.NewKeyRange([]byte("a"), []byte("a1")), 100)
+	re.Equal(int64(0), rgs.Count)
+	rgs,_=client.GetRegionsByKeyRange(env.ctx, pd.NewKeyRange([]byte("a1"), []byte("a3")), 100)
+	re.Equal(int64(2), rgs.Count)
+	rgs,_=client.GetRegionsByKeyRange(env.ctx, pd.NewKeyRange([]byte("a2"), []byte("b")), 100)
+	re.Equal(int64(1), rgs.Count)
```
Comment (JmPotato) [inline tests/integrations/client/http_client_test.go:212]: Better add test cases to test the empty start and end keys.

Comment (Smityz) [reply tests/integrations/client/http_client_test.go:212]: done

**Review by JmPotato — APPROVED** [review]
